### PR TITLE
chore: C4 event updates and normalization (SC-2034)

### PR DESCRIPTION
**Pull request by deluca-mike**
# Description

Rectify event-related issues raised by C4 and and normalize event indexing, cosmetics, and comments code-wide.

Clubhouse: https://app.clubhouse.io/maplefinance/story/2034/maple-core-c4-events-updates

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [X] Have any features changed or been added? If yes, outline below.
- [X] Have any events changed or been added? If yes, outline below.
- [X] Has all documentation been updated?

# Changelog

- Comment indicates that FundsWithdrawn event should be emitted only when _withdrawableDividend > 0  (https://github.com/code-423n4/2021-04-maple-findings/issues/96)
- No event is emitted (https://github.com/code-423n4/2021-04-maple-findings/issues/8)
- Missing event for critical operation of setAdmin change for Protocol admin in MapleGlobals.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/38)
- Missing event for critical operation of Pool Delegate validity change in MapleGlobals.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/39)
- Missing event for critical operation of setAdmin change in Pool.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/69)
- Missing index on events (https://github.com/code-423n4/2021-04-maple-findings/issues/86)
- Function triggerDefault should call _emitBalanceUpdateEventForCollateralLocker (https://github.com/code-423n4/2021-04-maple-findings/issues/99)
- Missing event for critical operation of new Collateral locker creation in CollateralLockerFactory.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/27)
- Missing event for critical operation of new Debt locker creation in DebtLockerFactory.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/29)
- Missing event for critical operation of new Funding locker creation in FundingLockerFactory.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/30)
- Missing event for critical operation of new Liquidity locker creation in LiquidityLockerFactory.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/31)
- Missing event for critical operation of setAdmin change in Loan.sol (https://github.com/code-423n4/2021-04-maple-findings/issues/33)
- Normalized event declaration spacing
- Added @dev comment for all events thrown in functions
- Cleaned up @dev comments in BasicFDT and ExtendedFDT
- Added indexing for all reasonably filterable event parameters

## Event Signature Changes

- `BasicFDT.PointsCorrectionUpdated(address *indexed* account, int256 pointsCorrection)`
- `ExtendedFDT.LossesCorrectionUpdated(address *indexed* account, int256 lossesCorrection)`
- `CollateralLockerFactory.CollateralLockerCreated(address indexed owner, address collateralLocker, address collateralAsset)` (new)
- `DebtLockerFactory.DebtLockerCreated(address indexed owner, address debtLocker, address loan)` (new)
- `FundingLockerFactory.FundingLockerCreated(address indexed owner, address fundingLocker, address liquidityAsset)` (new)
- `LiquidityLockerFactory.LiquidityLockerCreated(address indexed owner, address liquidityLocker, address liquidityAsset)` (new)
- `Loan.AdminSet(address admin, bool allowed)` (new)
- `MapleGlobals.AdminSet(address newAdmin)` (new)
-`MapleGlobals.PoolDelegateAllowed(address delegate, bool valid)` (new)
- ~~MapleTreasury.FundsTokenModified(address by, address newFundsToken)~~ (Removed)
- `MapleTreasury.GlobalsSet(address newGlobals)` (new)
- `Pool.AdminSet(address newAdmin, bool allowed)` (new)
- `StakeLocker.StakeDateUpdated(address *indexed* staker, uint256 stakeDate)`
- `Loan.BalanceUpdated(address *indexed* who, address *indexed* token, uint256 balance)`
- `StakeLocker.BalanceUpdated(address *indexed* who, address *indexed* token, uint256 balance)`
- `StakeLocker.AllowListUpdated(address *indexed* staker, bool status)`
- `StakeLocker.StakeDateUpdated(address *indexed*  staker, uint256 stakeDate)`
- `StakeLocker.Stake(uint256 amount, address *indexed*  staker)`
- `StakeLocker.Unstake(uint256 amount, address *indexed* staker)`
- `StakeLockerFactory.StakeLockerCreated(address owner, address *indexed* owner, address stakeLocker, address stakeAsset, address liquidityAsset, string name, string symbol)`

### CODE REVIEW

File: contracts/CollateralLockerFactory.sol [hunk 1]
```diff
@@ -13,15 +13,20 @@ contract CollateralLockerFactory {
 
     uint8 public constant factoryType = 0;  // i.e FactoryType::COLLATERAL_LOCKER_FACTORY
 
+    event CollateralLockerCreated(address owner, address collateralLocker, address collateralAsset);
```
Comment (satyamakgec) [inline contracts/CollateralLockerFactory.sol:16]: it is better to `indexed` the owner, So offchain team can filter on the basis of `owner`

File: contracts/DebtLockerFactory.sol [hunk 2]
```diff
@@ -11,15 +11,20 @@ contract DebtLockerFactory {
 
     uint8 public constant factoryType = 1;  // i.e LockerFactoryTypes::DEBT_LOCKER_FACTORY
 
+    event DebtLockerCreated(address owner, address debtLocker, address loan);
```
Comment (satyamakgec) [inline contracts/DebtLockerFactory.sol:14]: ditto

File: contracts/FundingLockerFactory.sol [hunk 3]
```diff
@@ -13,15 +13,20 @@ contract FundingLockerFactory {
 
     uint8 public constant factoryType = 2;  // i.e FactoryType::FUNDING_LOCKER_FACTORY
 
+    event FundingLockerCreated(address owner, address fundingLocker, address liquidityAsset);
```
Comment (satyamakgec) [inline contracts/FundingLockerFactory.sol:16]: ditto

File: contracts/LiquidityLockerFactory.sol [hunk 4]
```diff
@@ -11,15 +11,20 @@ contract LiquidityLockerFactory {
 
     uint8 public constant factoryType = 3;        // i.e LockerFactoryTypes::LIQUIDITY_LOCKER_FACTORY
 
+    event LiquidityLockerCreated(address owner, address liquidityLocker, address liquidityAsset);
```
Comment (satyamakgec) [inline contracts/LiquidityLockerFactory.sol:14]: ditto

File: contracts/MapleGlobals.sol [hunk 5]
```diff
@@ -55,16 +54,18 @@ contract MapleGlobals {
...
+    event                 BalancerPoolSet(address balancerPool,             bool valid);
     event              PendingGovernorSet(address pendingGovernor);
     event                GovernorAccepted(address governor);
-    event                 GlobalsParamSet(bytes32 indexed which, uint256 value);
-    event               GlobalsAddressSet(bytes32 indexed which, address addr);
+    event                 GlobalsParamSet(bytes32 indexed which,            uint256 value);
+    event               GlobalsAddressSet(bytes32 indexed which,            address addr);
     event                  ProtocolPaused(bool pause);
+    event                        AdminSet(address newAdmin);
+    event             PoolDelegateAllowed(address delegate,                 bool valid);
```
Comment (lucas-manuel) [inline contracts/MapleGlobals.sol:68]: I think I actually prefer just leaving the parameters normally spaced in the case of events.

File: contracts/Loan.sol [hunk 6]
```diff
@@ -83,10 +83,12 @@ contract Loan is FDT, Pausable {
     uint256 public defaultSuffered;    // Difference between `amountRecovered` and `principalOwed` after liquidation
     uint256 public liquidationExcess;  // If `amountRecovered > principalOwed`, amount of liquidityAsset that is to be returned to borrower
 
-    event LoanFunded(uint256 amtFunded, address indexed _fundedBy);
-    event BalanceUpdated(address who, address token, uint256 balance);
-    event Drawdown(uint256 drawdownAmt);
+    event       LoanFunded(uint256 amtFunded,   address indexed _fundedBy);
```
Comment (satyamakgec) [inline contracts/Loan.sol:86]: extra space in `amtFunded, address indexed`

File: contracts/MapleGlobals.sol [hunk 7]
```diff
@@ -144,6 +150,7 @@ contract MapleGlobals {
 
     /**
         @dev Update the allowed Uniswap slippage percentage, in basis points. Only the Governor can call this function.
+        @dev It emits a `GlobalsParamSet` event.
```
Comment (lucas-manuel) [inline contracts/MapleGlobals.sol:153]: Thanks for adding these!

File: contracts/MplRewards.sol [hunk 8]
```diff
@@ -31,13 +31,13 @@ contract MplRewards is Ownable {
...
-    event Withdrawn(address indexed user, uint256 amount);
-    event RewardPaid(address indexed user, uint256 reward);
+    event            RewardAdded(uint256 reward);
+    event                 Staked(address indexed user, uint256 amount);
+    event              Withdrawn(address indexed user, uint256 amount);
+    event             RewardPaid(address indexed user, uint256 reward);
     event RewardsDurationUpdated(uint256 newDuration);
-    event Recovered(address token, uint256 amount);
-    event PauseChanged(bool isPaused);
+    event              Recovered(address token,        uint256 amount);
```
Comment (lucas-manuel) [inline contracts/MplRewards.sol:39]: Same here, lets use normal spacing

File: contracts/Pool.sol [hunk 9]
```diff
@@ -56,23 +56,25 @@ contract Pool is PoolFDT {
...
+    event              Claim(address indexed loan,     uint256 interest,      uint256 principal, uint256 fee);
+    event     BalanceUpdated(address indexed who,      address indexed token, uint256 balance);
+    event    LPStatusChanged(address indexed user,     bool status);
+    event    LiquidityCapSet(uint256 newLiquidityCap);
+    event    LockupPeriodSet(uint256 newLockupPeriod);
+    event      StakingFeeSet(uint256 newStakingFee);
+    event   PoolStateChanged(State state);
+    event           Cooldown(address indexed lp,       uint256 cooldown);
+    event PoolOpenedToPublic(bool isOpen);
+    event           AdminSet(address newAdmin,         bool allowed);
```
Comment (lucas-manuel) [inline contracts/Pool.sol:69]: Normal spacing

File: contracts/MapleGlobals.sol [hunk 10]
```diff
@@ -55,16 +54,18 @@ contract MapleGlobals {
     mapping(address => mapping(address => bool)) public validSubFactories;   // Mapping of valid sub factories
 
     event                     Initialized();
-    event              CollateralAssetSet(address asset, uint256 decimals, string symbol, bool valid);
-    event               LiquidityAssetSet(address asset, uint256 decimals, string symbol, bool valid);
-    event                       OracleSet(address asset, address oracle);
+    event              CollateralAssetSet(address asset,                    uint256 decimals, string symbol, bool valid);
+    event               LiquidityAssetSet(address asset,                    uint256 decimals, string symbol, bool valid);
+    event                       OracleSet(address asset,                    address oracle);
```
Comment (satyamakgec) [inline contracts/MapleGlobals.sol:59]: I personally don't think it is necessary (space)

File: contracts/StakeLocker.sol [hunk 11]
```diff
@@ -32,13 +32,13 @@ contract StakeLocker is StakeLockerFDT, Pausable {
...
-    event    StakeDateUpdated(address staker, uint256 stakeDate);
+    event      BalanceUpdated(address indexed who,    address indexed token,   uint256 balance);
+    event    AllowListUpdated(address indexed staker, bool status);
+    event    StakeDateUpdated(address indexed staker, uint256 stakeDate);
     event LockupPeriodUpdated(uint256 lockupPeriod);
     event            Cooldown(address indexed staker, uint256 cooldown);
-    event               Stake(uint256 amount, address staker);
-    event             Unstake(uint256 amount, address staker);
+    event               Stake(uint256 amount,         address indexed staker);
+    event             Unstake(uint256 amount,         address indexed staker);
```
Comment (lucas-manuel) [inline contracts/StakeLocker.sol:41]: Normal spacing

File: contracts/StakeLocker.sol [hunk 12]
```diff
@@ -32,13 +32,13 @@ contract StakeLocker is StakeLockerFDT, Pausable {
 
     bool public openToPublic;  // Boolean opening StakeLocker to public for staking BPTs
 
-    event      BalanceUpdated(address who, address token, uint256 balance);
-    event    AllowListUpdated(address staker, bool status);
-    event    StakeDateUpdated(address staker, uint256 stakeDate);
+    event      BalanceUpdated(address indexed who,    address indexed token,   uint256 balance);
```
Comment (satyamakgec) [inline contracts/StakeLocker.sol:35]: extra space not needed

File: contracts/library/PoolLib.sol [hunk 13]
```diff
@@ -22,9 +22,9 @@ library PoolLib {
     uint256 public constant WAD         = 10 ** 18;
     uint8   public constant DL_FACTORY  = 1;         // Factory type of `DebtLockerFactory`
 
-    event         LoanFunded(address indexed loan, address debtLocker, uint256 amountFunded);
-    event DepositDateUpdated(address indexed lp, uint256 depositDate);
-    event           Cooldown(address indexed lp, uint256 cooldown);
+    event         LoanFunded(address indexed loan, address debtLocker,   uint256 amountFunded);
```
Comment (satyamakgec) [inline contracts/library/PoolLib.sol:25]: extra space

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
LGTM, just formatting

**Review by satyamakgec — COMMENTED** [review]
Did you cover all contracts for an event comments?

File: contracts/MapleGlobals.sol (hunk 5 above)
Comment (deluca-mike) [reply contracts/MapleGlobals.sol:68]: Will undo.

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**